### ISSUE TITLE
Problem with Data-Sparse Cells

### ISSUE THREAD

**Issue by Genajoin** [Genajoin/PyParaglide#2]
**Problem:** The model learns incorrect patterns for cells with few or no flight records due to data collection issues, not because flying is impossible there.

### Current Behavior

When a cell has zero or few flights, the model interprets this as "non-flyable weather" (label = 0), even though the actual reason may be:
- Poor data collection
- Low pilot population in the area
- Geographic access issues

This creates **label noise** and **selection bias** in the training data.

### Example

```
Cell A (rich data):
- Weather: wind=5m/s, sun=1
- Flights: 100
- Label: 1 (flyable) ✓

Cell B (sparse data):
- Weather: wind=5m/s, sun=1 (same conditions!)
- Flights: 0 (data collection issue)
- Label: 0 (not flyable) ✗ WRONG!
```

The model learns: "wind=5m/s, sun=1 → not flyable" for Cell B!

### Impact

1. **Poor generalization** to new regions with sparse data
2. **Systematic underestimation** of flyability in data-sparse areas
3. **Model confusion** between "no data" and "not flyable"

## Root Cause Analysis

In [`dataset.py:292-304`](src/pyparaglide/data/dataset.py#L292-L304):

```python
if len(daycell) > 0:
    res[0][res_line, 0] = 1.0  # Has flights → label 1
else:
    res[0][res_line, 0] = 0.0  # No flights → label 0  ← PROBLEM!
```

The code doesn't distinguish between:
- "Weather is bad, no one flew" (correct label = 0)
- "Weather is good, but data not collected" (label should be unknown)

### Solution 1: Cell Filtering (Quick Fix)

```python
MIN_FLIGHTS_PER_CELL = 20

# Filter cells with insufficient data from TRAINING
valid_cells = [c for c in cells if cell_flight_counts[c] >= MIN_FLIGHTS_PER_CELL]
```

**Pros:** Simple, fast to implement
**Cons:** Loses potentially useful data

### Solution 2: Sample Weighting (Recommended)

```python
# Weight samples based on cell data richness
sample_weights = {
    'rich_cell': 1.0,    # Full weight
    'poor_cell': 0.3,    # Reduced weight
}

loss = weighted_binary_crossentropy(y_true, y_pred, sample_weights)
```

**Pros:** Keeps all data, reduces noise impact
**Cons:** Requires tuning weights

### Solution 3: Transfer Learning from Similar Cells

```python
# For cells with sparse data, transfer patterns from similar cells
# Based on weather patterns or geographic proximity

def find_similar_cells(target_cell, metric='weather_pattern'):
    # Find N cells with similar weather distributions
    pass

def transfer_labels(target_cell, similar_cells):
    # Use labels from similar cells for unlabeled days
    pass
```

**Pros:** Best use of available data
**Cons:** Complex implementation

## Tasks

- [ ] Analyze flight distribution per cell
- [ ] Determine threshold for "data-sparse" cells
- [ ] Implement cell filtering or sample weighting
- [ ] Add metrics to track cell data quality
- [ ] Test model on held-out sparse cells

## Additional Context

See discussion in:
- [`layers.py`](src/pyparaglide/models/layers.py) - FlyabilityBlock architecture
- [`model_cells.py`](src/pyparaglide/models/model_cells.py) - CELLS model
- [`dataset.py`](src/pyparaglide/data/dataset.py) - Data loading

---

## Labels Suggestion
`bug`, `data-quality`, `enhancement`, `machine-learning`
